### PR TITLE
vendor: Add

### LINKED ISSUE #5: [go] Establish vendor dependencies for go
I am trying to build manager and other binaries from source code but it seems that we do not import all vendors. I think we should use dep or glide to establish vendor dependencies.

### PR DESCRIPTION
**Pull request by gaocegege**
Close #5 

Using dep to install all vendor dependencies.

/assign @YujiOshima

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+
+[[constraint]]
+  name = "github.com/docker/docker"
+  version = "1.13.1"
```
Comment (AkihiroSuda) [inline Gopkg.toml]: Note: this version is no longer maintained. I'd suggest syncing with the master.

Comment (gaocegege) [reply Gopkg.toml]: @AkihiroSuda Thanks for your advice, I will update it.

**Comment by gaocegege** [comment]
@AkihiroSuda 

Addressed. PTAL

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+
+[[constraint]]
+  name = "github.com/docker/docker"
+  branch = "master"
```
Comment (AkihiroSuda) [inline Gopkg.toml]: nit: I guess you don't need to retain this constraint entry for the master branch

**Comment by YujiOshima** [comment]
Thank you for great work!

Each docker image is still using `go get` packages.
We should update Dockerfiles to use the vendored packages.

**Comment by YujiOshima** [comment]
@gaocegege Absolutely, It can reduce image size so much and accelerate images build.

**Comment by gaocegege** [comment]
I opened an issue to keep track of the process https://github.com/kubeflow/hp-tuning/issues/14

**Comment by YujiOshima** [comment]
Thanks! 
/lgtm

**Pull request merged**